src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,90 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> currentNode = table[getIndex((key))];
+        if (table[getIndex(key)] == null) {
+            table[getIndex((key))] = new Node<>(Objects.hashCode(key), key, value, null);
+            size++;
+        }
+        while (currentNode != null) {
+            if (Objects.equals(key, currentNode.key)) {
+                currentNode.value = value;
+                break;
+            } else if (currentNode.next == null) {
```

Comment:
```suggestion
            } 
            if (currentNode.next == null) {
```

Reply:
I agree that we may try to replace this like Max suggests